Traducción (es) de index.html a Markdown index.md (#4251)
mdn/translated-content: files/es/web/accessibility/aria/index.md

[La primera regla de ARIA](https://www.w3.org/TR/using-aria/#rule1) es <q>"Si puede usar un elemento o atributo HTML nativo con la semántica y el comportamiento que necesita ya integrado, en lugar de reutilizar un elemento y agregar un rol, estado o propiedad de ARIA para hacerlo accesible, hágalo".</q>

> **Nota:** Hay un dicho "No utilizar ARIA es mejor que utilizar una mala ARIA". En [la encuesta de WebAim de más de un millón de páginas de inicio](https://webaim.org/projects/million#aria), encontraron que las páginas de inicio con ARIA presentes promediaron un 41 % más de errores detectados que aquellas sin ARIA. Si bien ARIA está diseñado para hacer que las páginas web sean más accesibles, si se usa incorrectamente, puede hacer más daño que bien.

Aquí está el marcado para una barra de progreso:

```html
<div
  id="percent-loaded"
  role="progressbar"
  aria-valuenow="75"
  aria-valuemin="0"
  aria-valuemax="100"
></div>
```

> Esta barra de progreso se construye usando un {{HTMLElement("div")}}, que no tiene significado. Incluimos roles y propiedades de ARIA para agregar significado. En este ejemplo, el atributo [`role="progressbar"`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) informa al navegador que este elemento es en realidad un widget de barra de progreso impulsado por JavaScript. [`aria-valuemin`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-valuemin) y [`aria-valuemax`](/en-US/docs/Web/Accessibility/ARIA/ Attributes/aria-valuemax) especifican los valores mínimo y máximo para la barra de progreso, y [`aria-valuenow`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-valuenow) describe el actual estado del mismo y por lo tanto debe mantenerse actualizado con JavaScript.

Reviewer: ```suggestion
Esta barra de progreso se construye usando un {{HTMLElement("div")}}, que no tiene significado. Incluimos roles y propiedades de ARIA para agregar significado. En este ejemplo, el atributo [`role="progressbar"`](/es/docs/Web/Accessibility/ARIA/Roles/progressbar_role) informa al navegador que este elemento es en realidad un widget de barra de progreso impulsado por JavaScript. [`aria-valuemin`](/es/docs/Web/Accessibility/ARIA/Attributes/aria-valuemin) y [`aria-valuemax`](/es/docs/Web/Accessibility/ARIA/ Attributes/aria-valuemax) especifican los valores mínimo y máximo para la barra de progreso, y [`aria-valuenow`](/es/docs/Web/Accessibility/ARIA/Attributes/aria-valuenow) describe el actual estado del mismo y por lo tanto debe mantenerse actualizado con JavaScript.
```

Además de colocarlos directamente en el marcado, los atributos ARIA se pueden agregar al elemento y actualizar dinámicamente usando un código JavaScript como este:

```js
// Find the progress bar <div> in the DOM.
var progressBar = document.getElementById("percent-loaded");

// Set its ARIA roles and states,
// so that assistive technologies know what kind of widget it is.
progressBar.setAttribute("role", "progressbar");
progressBar.setAttribute("aria-valuemin", 0);
progressBar.setAttribute("aria-valuemax", 100);

// Create a function that can be called at any time to update
// the value of the progress bar.
function updateProgress(percentComplete) {
  progressBar.setAttribute("aria-valuenow", percentComplete);
}

> ```

Author: Ok, me parece una buena decisión, voy a traducir los comentarios.

> ```

Reviewer 2: > Ok, me parece una buena decisión, voy a traducir los comentarios.

gracias dariofin!

Habría sido mucho más simple usar el elemento nativo {{HTMLElement('progress')}} en su lugar:

```HTML
<progress id="percent-loaded" value="75" max="100">75 %</progress>
```

> **Nota:** El atributo `min` no está permitido para el elemento {{HTMLElement('progress')}}; su valor mínimo es siempre `0`.

> **Nota:** Los elementos de referencia HTML ({{HTMLElement("main")}}, {{HTMLElement("header")}}, {{HTMLElement("nav")}} etc.) tienen roles ARIA implícitos, por lo que no es necesario duplicarlos.

## Soporte

Como cualquier otra tecnología web, existen diversos grados de soporte para ARIA. El soporte se basa en el sistema operativo y el navegador que se utiliza, así como en el tipo de tecnología de asistencia que interactúa con él. Además, la versión del sistema operativo, el navegador y la tecnología de asistencia son factores que contribuyen. Es posible que las versiones de software anteriores no admitan ciertas funciones de ARIA, solo tengan compatibilidad parcial o informen incorrectamente su funcionalidad.

> También es importante reconocer que algunas personas que confían en la tecnología de asistencia son reacias a actualizar su software por temor a perder la capacidad de interactuar con su computadora y navegador. Debido a esto, es importante [usar elementos HTML semánticos](/en-US/docs/Learn/Accessibility/HTML) siempre que sea posible, ya que el HTML semántico es mucho más compatible con la tecnología de asistencia.

Reviewer: ```suggestion
También es importante reconocer que algunas personas que confían en la tecnología de asistencia son reacias a actualizar su software por temor a perder la capacidad de interactuar con su computadora y navegador. Debido a esto, es importante [usar elementos HTML semánticos](/es/docs/Learn/Accessibility/HTML) siempre que sea posible, ya que el HTML semántico es mucho más compatible con la tecnología de asistencia.
```

También es importante probar su ARIA creado con tecnología de asistencia real. Así como los emuladores y simuladores de navegador no son una solución efectiva para probar el soporte completo, las soluciones de tecnología de asistencia de emuladas no son suficientes para garantizar completamente la funcionalidad.

## Tutoriales

> - [Introducción a ARIA](/en-US/docs/Web/Accessibility/An_overview_of_accessible_web_applications_and_widgets)

Reviewer: ```suggestion
- [Introducción a ARIA](/es/docs/Web/Accessibility/An_overview_of_accessible_web_applications_and_widgets)
```

  - : Una introducción rápida a cómo hacer que el contenido dinámico sea accesible con ARIA. Véase también el clásico [introducción a ARIA de Gez Lemon](https://dev.opera.com/articles/view/introduction-to-wai-aria/), de 2008.
- [Videos de lectores de pantalla usando ARIA](https://zomigi.com/blog/videos-of-screen-readers-using-aria-updated/)

>   - : vea ejemplos reales y simplificados de toda la web, incluidos videos de ARIA "antes" y "después".

Reviewer 2: ```suggestion
  - : Vea ejemplos reales y simplificados de toda la web, incluidos videos de ARIA "antes" y "después".
```

- [Usando ARIA](https://w3c.github.io/using-aria/)
  - : Una guía práctica para desarrolladores. Sugiere qué atributos ARIA usar en elementos HTML. Las sugerencias se basan en las realidades de la implementación.

> ## Simple ARIA enhancements

Reviewer 2: ```suggestion
## Mejoras simples de ARIA
```

- [Mejora de la navegación de páginas con referencias de ARIA](https://www.paciellogroup.com/blog/2013/02/using-wai-aria-landmarks-2013/)
  - : Una buena introducción al uso de los puntos de referencia de ARIA para mejorar la navegación de la página web para los usuarios de lectores de pantalla. [Consulte también las notas de implementación históricas de ARIA] (https://www.paciellogroup.com/blog/2011/07/html5-accessibility-chops-aria-landmark-support/) y ejemplos en sitios reales (actualizado en julio de 2011 ).

> - [Mejorando la accesibilidad de los formularios](/en-US/docs/Web/Accessibility/ARIA/forms)

Reviewer 2: ```suggestion
- [Mejorando la accesibilidad de los formularios](/es/docs/Web/Accessibility/ARIA/forms)
```

  - : ¡ARIA no es solo para contenido dinámico! Aprenda a mejorar la accesibilidad de los formularios HTML utilizando atributos ARIA adicionales.

## ARIA para widgets con scripts

> - [Escribir widgets JavaScript navegables por teclado](/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets)

Reviewer: ```suggestion
- [Escribir widgets JavaScript navegables por teclado](/es/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets)
```

  -: Elementos integrados como {{HTMLElement("input")}}, {{HTMLElement("button")}} etc. tienen accesibilidad de teclado integrado. Si 'falsifica' estos con {{HTMLElement("div")}}s y ARIA, debe asegurarse de que sus widgets sean accesibles desde el teclado.

> - [Regiones en vivo](/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions)

Reviewer: ```suggestion
- [Regiones en vivo](/es/docs/Web/Accessibility/ARIA/ARIA_Live_Regions)
```

Author: @Vallejoanderson aquí veo un problema, el enlace sugerido aún no existe, observa el mensaje devuelto: 

Sorry, the page /es/docs/Web/Accessibility/ARIA/ARIA_Live_Regions could not be found.
![Captura de pantalla 2022-03-09 a las 17 26 45 p  m](https://user-images.githubusercontent.com/61764276/157530130-993105db-fdc1-49cf-ae27-88346d251d3e.png)

  - : Las regiones activas brindan sugerencias a los lectores de pantalla sobre cómo manejar los cambios en el contenido de una página.
- [Uso de regiones en vivo de ARIA para anunciar cambios de contenido](https://www.freedomscientific.com/Training/Surfs-up/AriaLiveRegions.htm)
  - : Un breve resumen de las regiones en vivo, por los creadores del software de lectura de pantalla JAWS. Las regiones en vivo también son compatibles con NVDA con Firefox y VoiceOver con Safari.

## Referencias

> - [Funciones ARIA](/en-US/docs/Web/Accessibility/ARIA/Roles)

Reviewer: ```suggestion
- [Funciones ARIA](/es/docs/Web/Accessibility/ARIA/Roles)
```

  - : Páginas de referencia que cubren todos los roles de WAI-ARIA discutidos en MDN.
- [Estados y propiedades de ARIA](/en-US/docs/Web/Accessibility/ARIA/Attributes)
  - : Páginas de referencia que cubren todos los estados y propiedades de WAI-ARIA discutidos en MDN.rted por NVDA con Firefox y VoiceOver con Safari.

## Esfuerzos de estandarización

- [Especificación WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/)
  - : La propia especificación W3C.
- [Prácticas de creación de WAI-ARIA](https://www.w3.org/TR/wai-aria-practices-1.1/)
  - : Las mejores prácticas oficiales documentan la mejor manera de ARIA-ficar widgets e interacciones comunes. Un excelente recurso.

## Videos

Las siguientes charlas son una excelente manera de entender ARIA:

[¡ARIA, las API de accesibilidad y la codificación como si te importara un comino! – Leonie Watson](https://www.youtube.com/watch?v=qdB8SRhqvFc)

## Archivando errores

> [Archivar errores de ARIA en navegadores, lectores de pantalla y bibliotecas JavaScript.](/en-US/docs/Web/Accessibility/ARIA/How_to_file_ARIA-related_bugs)

Reviewer: Procuramos usar los enlaces a la version en espanol :)
```suggestion
[Archivar errores de ARIA en navegadores, lectores de pantalla y bibliotecas JavaScript.](/es/docs/Web/Accessibility/ARIA/How_to_file_ARIA-related_bugs)
```

## Temas relacionados

> [Accesibilidad](/es/docs/Web/Accessibility), [AJAX](/en-US/docs/Web/Guide/AJAX), [JavaScript](/en-US/docs/Web/JavaScript)

Reviewer 2: ```suggestion
[Accesibilidad](/es/docs/Web/Accessibility), [AJAX](/es/docs/Web/Guide/AJAX), [JavaScript](/es/docs/Web/JavaScript)
```

> [Accesibilidad](/en-US/docs/Web/Accessibility), [AJAX](/en-US/docs/Web/Guide/AJAX), [JavaScript](/en-US/docs/Web/JavaScript)

Reviewer: ```suggestion
[Accesibilidad](/es/docs/Web/Accessibility), [AJAX](/en-US/docs/Web/Guide/AJAX), [JavaScript](/en-US/docs/Web/JavaScript)
```

---
title: ARIA
slug: Web/Accessibility/ARIA
tags:
  - ARIA
  - Accessibility
  - HTML
  - Reference
  - WAI-ARIA
  translation_of: Web/Accessibility/ARIA

> ---

Reviewer 2: Hola @dariofin, espero que todo este muy bien por tu parte!

Solo estoy haciendo seguimiento al PR,
necesitas ayuda?
has considerado aplicar la sugerencia que hice?

Si tienes preguntas/inquietudes no dudes en compartirlas por este medio.  

Saludos  :pray:

> ---

Author: Hola @JuanVqz, todo muy bien por aquí, gracias.

Necesitaría ayuda con algunos conceptos pues soy novato dentro de github.
He visto algunas sugerencias de @Vallejoanderson que son interesantes y deberían ser aplicadas para mejorar el contenido, pero no se que acciones tomar para aplicar esas sugerencias ni tampoco estoy seguro si tengo los privilegios para autorizar esos cambios.

También he considerado útil la sugerencia que tu haz hecho, pero estoy en la misma situación que acabo de comentar, ignoro los procedimientos a realizar.

Pido disculpas por no tener la dinámica de funcionamiento ideal dentro de la plataforma, estoy esforzándome por aprender.

Saludos!

> ---

Reviewer 2: > Hola @JuanVqz, todo muy bien por aquí, gracias.
> 
> Necesitaría ayuda con algunos conceptos pues soy novato dentro de github. He visto algunas sugerencias de @Vallejoanderson que son interesantes y deberían ser aplicadas para mejorar el contenido, pero no se que acciones tomar para aplicar esas sugerencias ni tampoco estoy seguro si tengo los privilegios para autorizar esos cambios.
> 
> También he considerado útil la sugerencia que tu haz hecho, pero estoy en la misma situación que acabo de comentar, ignoro los procedimientos a realizar.
> 
> Pido disculpas por no tener la dinámica de funcionamiento ideal dentro de la plataforma, estoy esforzándome por aprender.
> 
> Saludos!

Hola @dariofin, no te preocupes! para eso estamos aquí, para **aprender y ayudar**.

Dado que eres el creador del Pull Request, creo que no tendrias problemas con los permisos.

Con respecto a las sugerencias que hizo @Vallejoanderson, digamos que son sugerencias dirigidas, es decir Vallejo ya hizo el codigo que deberias incluir en tu PR, como puedes ver en la siguiente imagen en el cuadro número 1 puedes observar el cambio en una ruta/enlace de `es-US` a `es` y esa es la **sugerencia**.

Ahora puedes ver en el cuadro número 2 un boton que dice `Commit sugerencia`, eso lo que hara es añadir el código que Vallejo sugirió al código de tu PR. En este caso como es algo que si queremos, por favor (si puedes ver el boton y hacerle click) te invito a que lo hagas con cada sugerencia que Vallejo realizó.
 

![image](https://user-images.githubusercontent.com/7331511/157449678-a0a062a6-c093-4155-bfd3-21aff535aaac.png)

Por ultimo pero no menos importante, que valiente eres al intentar contribuir cuando aun no estas comodo con GitHub :1st_place_medal: , si quieres tener más ayuda unete a nuestro chat en [telegram](https://t.me/+Dr6qKQCAepw4MjFj) o en [matrix](https://chat.mozilla.org/#/room/!cIEBGoIHFpsXNeVUwp:mozilla.org) para poder ayudarte más.
Saludos!

> ---

Author: @JuanVqz gracias por la paciencia y por tomarte el tiempo para guiarme en esta plataforma con claridad.

Te comento que seguí tu recomendación y fui haciendo "commit suggestion" a las sugerencias de @Vallejoanderson, pero en el proceso, me dí cuenta que yo no había chequeado que las urls funcionaran. En la siguiente sugerencia, lo hice y comprobé que justamente la url del enlace en español aún no existe, entonces se lo comenté a @Vallejoanderson e hice este comentario.

Por sentido común pienso que no debería aprobar enlaces rotos, a menos que la idea sea crear esos enlaces antes de publicar los cambios finales en la página, cosa que también me parece un poco desordenada, pero desconozco el procedimiento completo así que dejo planteada esta duda.

Saludos!

> ---

Reviewer 2: Que tal @dariofin, entiendo tu preocupación pero MDN (y directamente yari) que es el "core" de la página de MDN, por default, se encarga de hacer `fallback`, es decir, si no encuentra las rutas al español tomara las rutas en ingles, entonces no te preocupes que no existan las rutas en español.

Ahora, la explicación de por que preferimos que lo apuntes a español, si en un futuro algún contribuidor crea la página, ahora si tomará la ruta al español en lugar de caer en la versión en ingles.

Hace sentido? Si tienes más dudas no lo pienses, solo pregunta.

entonces, ya no tienes mas sugerencias, cierto? 

 ```mermaid
flowchart TD
  A[Click a un enlace `/es/javascript/algo`] --> B{Existe en español?};
  B -- Si --> C[Ver documentación en español]; 
  B -- No --> D[Ver documentación en ingles];

```

```mermaid
 
flowchart TD
  A[Click a un enlace `/en-US/javascript/algo`] --> B[Ver documentación en ingles];
```

> ---

Author: Excelente! la respuesta clarisima y bien por el core Yari que tiene el condicional para saber a donde dirigir al usuario! Voy a continuar entonces "aprobando" (no se si la expresión es correcta) los commit de @Vallejoanderson

> ---

Author: Bueno, siguiendo el PR, debajo tengo una advertencia en rojo sobre cambios "Changes requested", intenté tomar acciones haciendo un Review de esos cambios pero la advertencia sigue allí. 

Por otro lado, recibí un mail donde ennumera 9 "Flaws" consecuencia de enlaces rotos, que supongo es el core que me indica que redireccionará dichos enlaces rotos a su versión original en inglés.
![Captura de Pantalla 2022-03-09 a la(s) 22 09 01](https://user-images.githubusercontent.com/61764276/157566962-9d83fd12-d29d-4a53-90b8-3f35f713c387.png)

> ---

Reviewer 2: > Bueno, siguiendo el PR, debajo tengo una advertencia en rojo sobre cambios "Changes requested", intenté tomar acciones haciendo un Review de esos cambios pero la advertencia sigue allí.
> 
> Por otro lado, recibí un mail donde ennumera 9 "Flaws" consecuencia de enlaces rotos, que supongo es el core que me indica que redireccionará dichos enlaces rotos a su versión original en inglés. ![Captura de Pantalla 2022-03-09 a la(s) 22 09 01](https://user-images.githubusercontent.com/61764276/157566962-9d83fd12-d29d-4a53-90b8-3f35f713c387.png)

es correcto! MDN redireccionara a la versión en inglés. :+1:
